task4/service/src/main/java/com/epam/esm/service/exception/tag/TagNotFoundException.java

```diff
@@ -0,0 +1,19 @@
+package com.epam.esm.service.exception.tag;
+
+import com.epam.esm.service.exception.ServiceException;
+
+public class TagNotFoundException extends ServiceException {
```

Comment:
I think it's better to use the same exception for different entities but use different error codes.
E.g. NotFoundException("001") for Tag, NotFoundException("002") for Certificate ...